network_test_harness.go

```diff
@@ -0,0 +1,302 @@
+		             } else {
+			              break
+		             }
+	                   }
+	                   if finishResult.StrmID != strmID {
+		                t.Errorf("Expecting finishResult to have strmID: %v, but got %v", strmID, finishResult)
+	                   }
+            //NodeN sends TranscodeResponse to Node1
+	    case  "TranscodeResponse":
+                   setHandler(nodes[1])
+		   go func() {
+			   err := vn[*nodeCount-1].SendTranscodeResponse(peer.IDHexEncode(nodes[1].Identity), fmt.Sprintf("%v:%v", strmID, 0), map[string]string{"strmid1": "P240p30fps4x3", "strmid2": "P360p30fps4x3"})
+				if err != nil {
+					t.Errorf("Error sending transcode result: %v", err)
+				}
+		   }()
+		   select {
+			case r := <-rc:
+				if r["strmid1"] != "P240p30fps4x3" {
+					t.Errorf("Expecting %v, got %v", "P240p30fps4x3", r["strmid1"])
+				}
+				if r["strmid2"] != "P360p30fps4x3" {
+					t.Errorf("Expecting %v, got %v", "P360p30fps4x3", r["strmid2"])
+				}
+			case <-time.After(time.Second * 5):
+				t.Errorf("Timed out")
+		   }
+	   //NodeN request MasterPlayList to Node1, node1 sends the MasterPlaylist to nodeN 
+	   case "GetMasterPlaylist":
+                setHandler(nodes[1])
+		mpl := m3u8.NewMasterPlaylist()
+		pl, _ := m3u8.NewMediaPlaylist(10, 10)
+		mpl.Append("test.m3u8", pl, m3u8.VariantParams{Bandwidth: 100000})
+		strmID := fmt.Sprintf("%vba1637fd2531f50f9e8f99a37b48d7cfe12fa498ff6da8d6b63279b4632101d5e8b1c872c", peer.IDHexEncode(vn[1].NetworkNode.Identity))
+
+		//node1 Updates Playlist
+		if err := vn[1].UpdateMasterPlaylist(strmID, mpl); err != nil {
+			t.Errorf("Error updating master playlist")
+		}
+
+		//nodeN Gets Playlist
+		mplc, err := vn[*nodeCount-1].GetMasterPlaylist(vn[1].GetNodeID(), strmID)
+		if err != nil {
+			t.Errorf("Error getting master playlist: %v", err)
+		}
+		select {
+		case r := <-mplc:
+			vars := r.Variants
+			glog.Infof("got: %v", r)
+			if len(vars) != 1 {
+				t.Errorf("Expecting 1 variants, but got: %v - %v", len(vars), r)
+			}
+		case <-time.After(time.Second * 3):
+			glog.Infof("n1 mplMap: %v", vn[1].mplMap)
+			t.Errorf("Timed out")
+		}
+
+	   //NodeN sends GetNodeStatus to Node1, node1 sends the NodeStatus to NodeN
+	   case "GetNodeStatus":
+                setHandler(nodes[1])
+		//Add a manifest
+		mpl := m3u8.NewMasterPlaylist()
+		pl, _ := m3u8.NewMediaPlaylist(10, 10)
+		mpl.Append("test.m3u8", pl, m3u8.VariantParams{Bandwidth: 100000})
+		vn[1].UpdateMasterPlaylist("testStrm", mpl)
+		sc, err := vn[*nodeCount-1].GetNodeStatus(vn[1].GetNodeID())
+		if err != nil {
+			t.Errorf("Error: %v", err)
+		}
+		status := <-sc
+		if len(status.Manifests) != 1 {
+			t.Errorf("Expecting 1 manifest, but got %v", status.Manifests)
+		}
+	}
+	  //check node status(subscribers,relayers..) 
+	for i:=0; i<*nodeCount; i++ {
+		 glog.Infof("node%v subsribers: %v relayers: %v",i, vn[i].subscribers,vn[i].relayers)
+	}
+
+}
+
+
+func setHandler(n *NetworkNode) {
```

Comment:
Can we use `basic_network->SetupProtocol` instead of writing new handlers?

Reply:
Yes I think so